src/appdistribution/client.ts

```diff
@@ -1,182 +1,158 @@
 }
 
-/** Enum for app_view parameter for getApp requests */
-export enum AppView {
-  BASIC = "BASIC",
-  FULL = "FULL",
+export interface BatchRemoveTestersResponse {
+  emails: string[];
 }
 
 /**
- * Proxies HTTPS requests to the App Distribution server backend.
+ * Makes RPCs to the App Distribution server backend.
  */
 export class AppDistributionClient {
-  static MAX_POLLING_RETRIES = 60;
-  static POLLING_INTERVAL_MS = 2000;
+  appDistroV2Client = new Client({
+    urlPrefix: api.appDistributionOrigin,
+    apiVersion: "v1",
+  });
 
-  constructor(private readonly appId: string) {}
-
-  async getApp(appView = AppView.BASIC): Promise<AppDistributionApp> {
-    const apiResponse = await api.request("GET", `/v1alpha/apps/${this.appId}?appView=${appView}`, {
+  async getAabInfo(appName: string): Promise<AabInfo> {
+    const apiResponse = await api.request("GET", `/v1/${appName}/aabInfo`, {
       origin: api.appDistributionOrigin,
       auth: true,
     });
 
     return _.get(apiResponse, "body");
   }
 
-  async uploadDistribution(distribution: Distribution): Promise<string> {
-    const apiResponse = await api.request("POST", `/app-binary-uploads?app_id=${this.appId}`, {
+  async uploadRelease(appName: string, distribution: Distribution): Promise<string> {
+    const apiResponse = await api.request("POST", `/upload/v1/${appName}/releases:upload`, {
       auth: true,
       origin: api.appDistributionOrigin,
       headers: {
-        "X-APP-DISTRO-API-CLIENT-ID": pkg.name,
-        "X-APP-DISTRO-API-CLIENT-TYPE": distribution.platform(),
-        "X-APP-DISTRO-API-CLIENT-VERSION": pkg.version,
-        "X-GOOG-UPLOAD-FILE-NAME": distribution.getFileName(),
-        "X-GOOG-UPLOAD-PROTOCOL": "raw",
+        "X-Goog-Upload-File-Name": distribution.getFileName(),
+        "X-Goog-Upload-Protocol": "raw",
         "Content-Type": "application/octet-stream",
       },
       data: distribution.readStream(),
       json: false,
     });
 
-    return _.get(JSON.parse(apiResponse.body), "token");
-  }
-
-  async pollUploadStatus(binaryName: string, retryCount = 0): Promise<string> {
-    const uploadStatus = await this.getUploadStatus(binaryName);
-    if (uploadStatus.status === UploadStatus.IN_PROGRESS) {
-      if (retryCount >= AppDistributionClient.MAX_POLLING_RETRIES) {
-        throw new FirebaseError(
-          "it took longer than expected to process your binary, please try again",
-          { exit: 1 }
-        );
-      }
-      await new Promise((resolve) =>
-        setTimeout(resolve, AppDistributionClient.POLLING_INTERVAL_MS)
-      );
-      return this.pollUploadStatus(binaryName, retryCount + 1);
-    } else if (uploadStatus.status === UploadStatus.SUCCESS) {
-      return uploadStatus.release.id;
-    } else {
-      throw new FirebaseError(
-        `error processing your binary: ${uploadStatus.message} (Code: ${uploadStatus.errorCode})`
-      );
-    }
+    return _.get(JSON.parse(apiResponse.body), "name");
```

Comment:
Why don't you have to unnest the `release` to inquire about the name?  I was expecting the JSON to return `{ "release": { "name": "<release-name>" } }` but this looks like you expect it not to have the top level `release` key.

Reply:
This [RPC is returning an `Operation`](http://google3/google/firebase/appdistro/v1/firebase_app_distribution.proto?l=47&rcl=391074892), not a `Release`. The `name` of the `Operation` is at the top level.